### PR TITLE
[HIPIFY][doc] Refactor documentation

### PR DESCRIPTION
**Pull request by SwRaw**
Refactoring all libraries to adhere to the documentation standards, which includes following home page format, index style, language improvements, link fixes, navigation, and overall content readability improvements.

### CODE REVIEW

**Comment by SwRaw** [comment]
@lpaoletti Please review. I have converted some files into rst however there are around 12 enormous files consisting supported CUDA APIs, which are still in markdown. I have not converted them as it is a very time-consuming process without making a difference.

File: docs/hipify-clang.rst [hunk 1]
```diff
@@ -1,32 +1,37 @@
+.. meta::
+   :description: Tools to automatically translate CUDA source code into portable HIP C++
+   :keywords: HIPIFY, ROCm, library, API, tool
```
Comment (emankov) [inline docs/hipify-clang.rst]: Here and in the rest of the changed files below:

`+` CUDA
`+` CUDA2HIP
`+` hipify-clang
`+` hipify-perl
`-` API

**Review by emankov — APPROVED** [review]
LGTM, except meta tags

File: docs/hipify-clang.rst (hunk 1 above)
Comment (SwRaw) [reply docs/hipify-clang.rst]: Done!

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+=====================
+HIPIFY Documentation
+=====================
+
+``hipify-clang`` and ``hipify-perl`` are tools that automatically translate CUDA source code into portable HIP C++.
+
+.. note::
+    
+    `hipify_torch <https://github.com/ROCmSoftwarePlatform/hipify_torch>`_ is a related tool that also translates CUDA source code into portable HIP C++. It was initially developed as part of the PyTorch project to cater to the project's unique requirements but was found to be useful for PyTorch-related projects and thus was released as an independent utility.
```
Comment (lpaoletti) [inline docs/index.rst:15]: We should change references to CUDA. They should be NVIDIA CUDA. 
This applies to the first occurrences on every page in this component.

File: docs/supported_apis.md [hunk 3]
```diff
@@ -15,4 +15,4 @@
 | DNN API         | [HIP DNN API](tables/CUDNN_API_supported_by_HIP.md)                  |       |        |
 | CUB API         | [HIP CUB API](tables/CUB_API_supported_by_HIP.md)                    |       |        |
```
Comment (lpaoletti) [inline docs/supported_apis.md:16]: The order is not consistent at the bottom of the list between the topic and the TOC. 
In addition, the items in the first column (CUDA) use names that are not the same as the TOC. For example:
CUBLAS
CURAND
CUDNN

image

File: docs/index.rst (hunk 2 above)
Comment (Rmalavally) [reply docs/index.rst:15]: Fixed.

File: docs/supported_apis.md (hunk 3 above)
Comment (Rmalavally) [reply docs/supported_apis.md:16]: Fixed.

**Comment by Rmalavally** [comment]
@emankov @lpaoletti Please review. Could you let me know if I need to change the base repository to amd-staging? Thanks so much.

https://github.com/SwRaw/HIPIFY/pull/1

**Pull request merged**